### PR TITLE
Epidermolysis Bullosa

### PR DESCRIPTION
**Pull request by Shadowmech88**
Adds new stage 3 virus symptom, epidermolysis bullosa.
This symptom causes the sufferer to take 10 brute damage to a random limb every time they bump something, something bumps them, or someone touches them (hugging, for example).
Note that the effect also triggers when people switch places with you via bumping on help intent.

I didn't add extra damage for being hit with objects since I figured that would be cruel, but I could if people want it.

:cl:
 * rscadd: Added epidermolysis bullosa as a new stage 3 virus symptom, causes damage every time something bumps or touches you.

### CODE REVIEW

**Comment by 9600bauds** [comment]
Hahah oh god add a cooldown to this or you will get instafuckingSHREDDED the moment someone in not-help intent runs into you

**Comment by despotate** [comment]
Instead of copy-pasting the same snowflake code across three different functions just for one symptom, why not make standardize an onTouch() function for effects, and then make the mob contact functions loop through all effects of all active viruses and run onTouch for those effects that have it?

**Comment by Fruktlimpa** [comment]
Imagine the old help-intent shuffle at an airlock and this person will go into crit faster than the clown can honk.

**Comment by Shadowmech88** [comment]
I'd rather forego the cooldown and bump the stage up to 3 or 4, if it's an issue.

**Comment by Shadowmech88** [comment]
Due to the fact that your `Bumped()` doesn't get called when someone switches places with you via bumping you on help intent, even though their `Bump()` *does* get called, I had to make an exception for mobs bumping each other in which the effects for the bumped mob get applied in the bumping mob's `Bump()`.

**Comment by 9600bauds** [comment]
Semantic satiation (also semantic saturation) is a psychological phenomenon in which repetition causes a word or phrase to temporarily lose meaning for the listener, who then perceives the speech as repeated meaningless sounds.

File: code/modules/virus2/effect/effect.dm [hunk 1]
```diff
@@ -461,22 +463,47 @@
 
 /datum/disease2/effect/butterfly_skin
 	name = "Epidermolysis Bullosa"
-	stage = 2
+	stage = 1
```
Comment (despotate) [inline code/modules/virus2/effect/effect.dm]: Diseases progress from stages 1 to 4 both in the code and in-game examination now - I think you meant this to be stage 4?

**Comment by despotate** [comment]
Here's an easy way to add a cooldown that also balances this effect if it's present in one of multiple viruses in a host:

Make a counter (not `count`) that increases every time the effect is `activate`d and decreases every time the person is `bump`ed, unless it's equal to 0 at which point nothing happens to the host and the counter doesn't decrease. Could also clamp the counter between 0 and 1 if you don't want to "stack" bump damage.

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -9,7 +9,30 @@
 	..()
 	if(istype(AM, /mob/living/carbon) && prob(10))
 		src.spread_disease_to(AM, "Contact")
-
+	if(virus2.len)
+		for(var/I in virus2)
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/carbon.dm]: Make this loop a proc on carbons and replace the 4 times you copypasted.

**Review by ComicIronic — DISMISSED** [review]

File: code/modules/mob/living/carbon/carbon.dm (hunk 2 above)
Comment (Shadowmech88) [reply code/modules/mob/living/carbon/carbon.dm]: Done.

**Comment by Shadowmech88** [comment]
* Epidermolysis bullosa is now a stage 3 symptom instead of stage 2, to compensate for the amount of damage it can cause.

**Comment by icantthinkofanameritenow** [comment]
https://en.wikipedia.org/wiki/Epidermolysis_bullosa

It's a real thing, who knew?

**Pull request merged**